testdata/cloudflare/cloudflare_turnstile_widget.yaml

```diff
@@ -0,0 +1,41 @@
+---
+version: 1
+interactions:
+- request:
+    body: ""
+    form: {}
+    headers:
+      Content-Type:
+      - application/json
+    url: https://api.cloudflare.com/client/v4/accounts/f037e56e89293a057740de681ac9abbe/challenges/widgets?page=1&per_page=25
+    method: GET
+  response:
+    body: |
+      {"result":[{"sitekey":"0x4AAAAAAAEg5sP3rwf91fe8","modified_on":"2023-05-08T11:49:41.862676Z","created_on":"2023-05-08T11:49:41.862676Z","mode":"managed","domains":["ungeek.fr"],"name":"test site","bot_fight_mode":false,"region":"world","offlabel":false},{"sitekey":"0x4AAAAAAAE0gwg0H1StXlOx","modified_on":"2023-05-17T02:08:18.399564Z","created_on":"2023-05-17T02:08:18.399564Z","mode":"managed","domains":["example.com"],"name":"My Terraform-managed widget","bot_fight_mode":false,"region":"world","offlabel":false},{"sitekey":"0x4AAAAAAAE2z4LbxEka5UBh","modified_on":"2023-05-17T19:11:26.151063Z","created_on":"2023-05-17T19:11:26.151063Z","mode":"managed","domains":["guerreiro.me"],"name":"My website","bot_fight_mode":false,"region":"world","offlabel":false}],"success":true,"errors":[],"messages":[],"result_info":{"page":1,"per_page":25,"count":3,"total_count":3,"total_pages":1}}
+    headers:
+      Content-Type:
+      - application/json
+      Vary:
+      - Accept-Encoding
+    status: 200 OK
+    code: 200
+    duration: ""
+- request:
+    body: ""
+    form: {}
+    headers:
+      Content-Type:
+      - application/json
+    url: https://api.cloudflare.com/client/v4/accounts/f037e56e89293a057740de681ac9abbe/challenges/widgets?page=2&per_page=25
```

Comment:
This second request shouldn't happen, but does. The `Done` function looks odd to me

```
// Done returns true for the last page and false otherwise.
func (p ResultInfo) Done() bool {
	return p.Page > 1 && p.Page > p.TotalPages
}
```

Should this be `p.Page >= p.TotalPages` instead?